### PR TITLE
Improve edge deletion performance

### PR DESCRIPTION
**Pull request by AviAvni**

### CODE REVIEW

File: src/execution_plan/ops/op_delete.c [hunk 1]
```diff
@@ -105,10 +105,13 @@ static void _DeleteEntities
 		// NOTE: delete edges before nodes
 		// required as a deleted node must be detached
 
+		MATRIX_POLICY policy = Graph_GetMatrixPolicy(op->gc->g);
+		Graph_SetMatrixPolicy(op->gc->g, SYNC_POLICY_NOP);
```
Comment (vityafx) [inline src/execution_plan/ops/op_delete.c:109]: Pardon me, please, for this question, but where can I read up on these flags? The enum it is in doesn't contain any documentation. I was wondering when I should use one flag or another and what they change when used.

Is this also a part of improving the performance?

File: src/util/simple_timer.c [hunk 2]
```diff
@@ -1,49 +1,60 @@
...
-        /* Linux has a very low resolution clock() function, so use the high
-           resolution clock_gettime instead.  May require -lrt */
+        // Linux has a very low resolution clock() function, so use the high
+        // resolution clock_gettime instead.  May require -lrt
         struct timespec t ;
         clock_gettime (CLOCK_MONOTONIC, &t) ;
         tic [0] = (double) t.tv_sec ;
         tic [1] = (double) t.tv_nsec ;
 
+    #elif defined ( __MACH__ ) && defined ( __APPLE__ )
```
Comment (vityafx) [inline src/util/simple_timer.c:39]: Is this also faster on macOS than using a C11 clock? In the comment below it is stated that the C11 clock on macOS also uses some high-resolution clock.

File: src/execution_plan/ops/op_delete.c (hunk 1 above)
Comment (AviAvni) [reply src/execution_plan/ops/op_delete.c:109]: you can look at the documentation at Graph_SetMatrixPolicy
this is the performance improvement don't sync the matrix during edge deletion

File: src/util/simple_timer.c (hunk 2 above)
Comment (AviAvni) [reply src/util/simple_timer.c:39]: this fixes a bug reporting incorrect timing on MacOS

File: src/execution_plan/ops/op_delete.c (hunk 1 above)
Comment (vityafx) [reply src/execution_plan/ops/op_delete.c:109]: Thanks !

File: src/util/simple_timer.c (hunk 2 above)
Comment (vityafx) [reply src/util/simple_timer.c:39]: Thanks for the explanation!

**Review by swilly22 — APPROVED** [review]

**Review by swilly22 — DISMISSED** [review]

File: src/graph/graph_hub.c [hunk 3]
```diff
@@ -171,23 +171,26 @@ uint DeleteNode
...
 	// add edge deletion operation to undo log
 	QueryCtx *query_ctx = QueryCtx_GetQueryCtx();
-	UndoLog_DeleteEdge(&query_ctx->undo_log, e);
+	uint count = array_len(edges);
+	for (uint i = 0; i < count; i++) {
+		UndoLog_DeleteEdge(&query_ctx->undo_log, edges + i);
 
-	if(GraphContext_HasIndices(gc)) {
-		_DeleteEdgeFromIndices(gc, e);
+		if(GraphContext_HasIndices(gc)) {
```
Comment (AviAvni) [inline src/graph/graph_hub.c:188]: call outside of loop

File: src/graph/graph.c [hunk 4]
```diff
@@ -1097,17 +1098,11 @@ int Graph_DeleteEdge
...
-	if(info != GrB_SUCCESS) {
-		return 0;
-	}
-
 	// an edge of type r has just been deleted, update statistics
 	GraphStatistics_DecEdgeCount(&g->stats, r, 1);
 
 	// single edge of type R connecting src to dest, delete entry
-	info = RG_Matrix_removeEntry(R, src_id, dest_id, ENTITY_GET_ID(e));
+	info = RG_Matrix_removeEntry(R, src_id, dest_id, ENTITY_GET_ID(e), &edge_id);
```
Comment (AviAvni) [inline src/graph/graph.c:1105]: add suffix UINT64

**Pull request merged**